From 32d6cec3be361c3d2228967d41b868b7b3b910c0 Mon Sep 17 00:00:00 2001
From: Titouan Christophe <titouan.christophe@railnova.eu>
Date: Sat, 12 Sep 2020 16:49:57 +0200
Subject: [PATCH] Use pkg-config to determine openssl LDFLAGS

Depending on its build configuration, openssl might not be compiled with
dynamic zlib support, and programs that link against openssl need to link
against zlib as well, as illustrated in this build failure:
http://autobuild.buildroot.net/results/ae9/ae946ca72238840b3eaa5fe823e8d620618f7462/

Signed-off-by: Titouan Christophe <titouan.christophe@railnova.eu>
[upstream status: https://github.com/angeloc/htpdate/pull/1]
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 09eea4a..21668a6 100644
--- a/Makefile
+++ b/Makefile
@@ -4,10 +4,11 @@ mandir = ${prefix}/share/man
 
 CC ?= gcc
 CFLAGS += -Wall -std=c99 -pedantic -O2
+PKG_CONFIG ?= pkg-config
 
 ifdef ENABLE_HTTPS
 CFLAGS += -DENABLE_HTTPS
-LDFLAGS += -lssl -lcrypto
+LDFLAGS += $(shell $(PKG_CONFIG) --libs openssl)
 endif
 
 INSTALL = install -c
-- 
2.25.3

